### ISSUE TITLE
fix: FluentDataGrid column resize doesn't work if grid contained in a relatively-positioned container

### ISSUE THREAD

**Issue by konvolution** [microsoft/fluentui-blazor#644]
# 🐛 Bug Report

If a `<FluentDataGrid/>` is placed inside a relatively-positioned element (for example), then column resizing doesn't work properly. This is because the pointer position is relative to the viewport, but the calculation of the left edge of the column is relative to the positioned element.

There are other minor issues with resizing. In particular, the resize cursor flickers as you're dragging it.

## 💻 Repro or Code Sample

It should be very easy to create a repro. In fact, you can just go to the component demo page and change one of the container elements to be `position: relative`.

## 🤔 Expected Behavior

When you click-and-drag a column resize handle, the right-hand edge of the column should track the horizontal position of the pointer (e.g. mouse cursor).

## 😯 Current Behavior

In the scenario described, when you initially press down on a column resize handle, the column becomes very wide, and it continues to be offset from the horizontal position of the pointer.

## 💁 Possible Solution

Here is some code that fixes the issue (including other minor issues, such as the resize cursor flickering):

```typescript
function enableColumnResizing(gridElement: GridElement) {
    const min = 50;
    const columns: {header: HTMLElement, size?: string }[] = [];
    let headerBeingResized: HTMLElement | undefined;
    let resizeHandle: HTMLElement | undefined;

    if (gridElement === null) {
        return;
    };

    gridElement.querySelectorAll('.column-header').forEach(header => {
        columns.push({
            header: <HTMLElement>header
        });

        const onPointerMove = (e: { clientX: number }) => requestAnimationFrame(() => {
            console.log(`onPointerMove${headerBeingResized ? '' : ' [not resizing]'}`);

            if (!headerBeingResized) {
                return;
            }

            const gridLeft = gridElement.getBoundingClientRect().left;
            const headerLocalLeft = headerBeingResized.getBoundingClientRect().left - gridLeft;
            const pointerLocalLeft = e.clientX - gridLeft;

            const width = pointerLocalLeft - headerLocalLeft;

            const column = columns.find(({ header }) => header === headerBeingResized)!;
            column.size = Math.max(min, width) + 'px';

            // Set initial sizes
            columns.forEach((column) => {
                if (column.size === undefined ) {
                    column.size = column.header.clientWidth + 'px';
                }
            });

            gridElement.gridTemplateColumns = columns
                .map(({ size }) => size)
                .join(' ');
        });

        const onPointerUp = () => {
            console.log('onPointerUp');

            headerBeingResized = undefined;
            resizeHandle = undefined;
        };

        const initResize = ({ target, pointerId }: { target: EventTarget | null, pointerId: number }) => {
            console.log('initResize');

            resizeHandle = <HTMLElement>target;
            headerBeingResized = <HTMLElement>resizeHandle.parentNode;

            resizeHandle.setPointerCapture(pointerId);
        };

        const dragHandle = <HTMLElement>header.querySelector('.col-width-draghandle');
        if (dragHandle) {
            dragHandle.addEventListener('pointerdown', initResize);
            dragHandle.addEventListener('pointermove', onPointerMove);
            dragHandle.addEventListener('pointerup', onPointerUp);
        }
    });
}
```

**Comment by vnbaaij** [comment]
Thank you! Unfortunately, I'm not getting a good resizing experience with this script. Probably because I had to convert it from TypeScript to JavaScript. Can you post a ready-to-paste js version of this code?

**Comment by konvolution** [comment]
Wow, that's fast. Thanks!

**Comment by gkizior** [comment]
> Thank you! ~Unfortunately, I'm not getting a good resizing experience with this script. Probably because I had to convert it from TypeScript to JavaScript. Can you post a ready-to-paste js version of this code?~
> 
> This will be implemented in v2.4.3 and v3

Do you have a timeline for full v3 release?

**Comment by vnbaaij** [comment]
Somewhere in the next few weeks, I hope
